### PR TITLE
feat(cdc): support more metadata columns for MySQL, PG and MongoDB

### LINKED ISSUE #16654: additional metadata when CDC via Debezium
### Is your feature request related to a problem? Please describe.

like this: https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/cdc-connectors/mysql-cdc/#available-metadata

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support more metadata columns for cdc table
- `database_name`, `schema_name`, `table_name` for PG and MySQL
- `collection_name` for MongoDB

Example:
```sql
CREATE TABLE person (
    id int,
    name varchar,
    email_address varchar,
    credit_card varchar,
    city varchar,
    PRIMARY KEY (id)
) INCLUDE TIMESTAMP AS commit_ts
INCLUDE DATABASE_NAME as database_name
INCLUDE SCHEMA_NAME as schema_name
INCLUDE TABLE_NAME as table_name
FROM pg_source TABLE 'public.person';
```

close #16654 

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Support more metadata columns for cdc table
- `database_name`, `schema_name`, `table_name` for PG and MySQL
- `collection_name` for MongoDB

</details>

### CODE REVIEW

File: proto/plan_common.proto [hunk 1]
```diff
@@ -215,6 +224,10 @@ message AdditionalColumn {
     AdditionalColumnHeader header_inner = 5;
     AdditionalColumnFilename filename = 6;
     AdditionalColumnHeaders headers = 7;
+    AdditionalDatabaseName database_name = 8;
```
Comment (BugenZhao) [inline proto/plan_common.proto:227]: Since they're all empty messages, can we use the same message type instead? Or directly use `google.protobuf.Empty`?

**Review by tabVersion — COMMENTED** [review]
the impl LGTM, just to confirm cdc source does not need the additional column to maintain its states so no need for compatible code?

File: proto/plan_common.proto (hunk 1 above)
Comment (tabVersion) [reply proto/plan_common.proto:227]: IIRC, prost has some trouble with google well-known types. 
& I use message structs for future extensibility on the column def, such as 

```
message AdditionalColumnHeader {
  string inner_field = 1;
  data.DataType data_type = 2;
}
```

we won't want to handle the changes inside the catalog in the future.

**Comment by StrikeW** [comment]
> the impl LGTM, just to confirm cdc source does not need the additional column to maintain its states so no need for compatible code?

No need compatible code for cdc sources and tables defined in previous version. User need to recreate their source or table to get the metadata column.

File: proto/plan_common.proto (hunk 1 above)
Comment (StrikeW) [reply proto/plan_common.proto:227]:
```
message AdditionalColumn {
  oneof column_type {
    AdditionalColumnKey key = 1;
    AdditionalColumnTimestamp timestamp = 2;
    AdditionalColumnPartition partition = 3;
    AdditionalColumnOffset offset = 4;
    AdditionalColumnHeader header_inner = 5;
    AdditionalColumnFilename filename = 6;
    AdditionalColumnHeaders headers = 7;
  }
}
```
I think we need different message type for the `oneof`.

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

**Review by tabVersion — APPROVED** [review]

**Pull request merged**